homeassistant/core.py

```diff
@@ -1355,6 +1358,10 @@ def path(self, *path: str) -> str:
             raise HomeAssistantError("config_dir is not set")
         return os.path.join(self.config_dir, *path)
 
+    def is_allowed_external_url(self, url: str) -> bool:
+        """Check if an external URL is allowed."""
+        return url in self.allowlist_external_urls
```

Comment:
This is not the right check. We need to check the path.

So if `http://example.com/bla/` is whitelisted, we allow `http://example.com/bla/paulus.png`

So we should check "starts_with"

Reply:
Doesn't that go against the idea of explicitly-defined URLs (that we established [here](https://github.com/home-assistant/architecture/issues/404#issuecomment-647077022))?

Reply:
It's okay to be paths under the urls that are allowed. We do the same for dirs.